### ISSUE TITLE
Heads up: Powershell snippet mixup variables with placeholders

### ISSUE THREAD

**Issue by egamma** [PowerShell/vscode-powershell#791]
The syntax has been been made stricter some time ago
https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax and we are now emitting warnings.

@daviwil these are the warnings I get in the latest insiders
```
The "Class"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "Constructor"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "Hidden Property"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "Property"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "Method"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "Enum"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "Parameter"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "do-until"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "do-while"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "while"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "for"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "for-reversed"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "function"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "if"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "elseif"-snippet very likely confuses snippet-variables and snippet-placeholders.
The "switch"-snippet very likely confuses snippet-variables and snippet-placeholders.

**Comment by daviwil** [comment]
Thanks for letting me know!  I've had reports recently (https://github.com/PowerShell/vscode-powershell/issues/767) of some of our snippet placeholders not working correctly, I'll fix those up in this week's release.

**Comment by rkeithhill** [comment]
Oops. Some (most?) of those are my bad.  Well, good that they are issue warnings now.

**Comment by daviwil** [comment]
Fixed with PR #795.
